lib/tokenize.cpp

```diff
@@ -6225,6 +6225,29 @@ void Tokenizer::simplifyHeadersAndUnusedTemplates()
     for (Token *tok = list.front(); tok; tok = tok->next()) {
         const bool isIncluded = (tok->fileIndex() != 0);
 
+        // Remove constructor initializer
+        if (isIncluded && !mSettings.checkHeaders && Token::Match(tok, "%name% (")) {
+            Token *start = tok->next()->link();
+            if (start) {
+                start = start->next();
+            }
+            if (start && start->str() == ":") {
+                const Token *next = start;
+                while (Token::Match(next, "[,:]")) {
```

Comment:
we expect that there is a name token after next right?

    while (Token::Match(next, "[,:] %name%"))

Reply:
Correct

Reply:
in my opinion it wouldn't hurt to add that more explicit check.